src/lib/message/L1Transaction.ts

```diff
@@ -104,13 +107,65 @@ export class L1TransactionReceipt implements TransactionReceipt {
     this.nitroReceipt = new nitro.L1TransactionReceipt(tx)
   }
 
-  private async looksLikeEthDeposit(message: classic.L1ToL2MessageReader) {
-    const inputs = await message.getInputs()
-    return (
-      inputs.maxGas.eq(0) &&
-      inputs.gasPriceBid.eq(0) &&
-      inputs.callDataLength.eq(0)
+  private async looksLikeEthDeposit(
+    gasLimit: BigNumber,
+    gasPrice: BigNumber,
+    callDataLength: number
+  ) {
+    return gasLimit.eq(0) && gasPrice.eq(0) && callDataLength === 0
+  }
+
+  private parseRetryableMessageData(eventData: string) {
+    // parse the event
+    // uint256(uint160(bytes20(destinationAddress))),
+    // uint256(0),
+    // msg.value,
+    // maxSubmissionCost,
+    // uint256(uint160(bytes20(destinationAddress))),
+    // uint256(uint160(bytes20(destinationAddress))),
+    // uint256(0),
+    // uint256(0),
+    // uint256(0),
+    // ""
+
+    // decode the data field - is been packed so we cant decode the bytes field this way
+    const parsed = defaultAbiCoder.decode(
+      [
+        'uint256',
+        'uint256',
+        'uint256',
+        'uint256',
+        'uint256',
+        'uint256',
+        'uint256',
+        'uint256',
+        'uint256', // data length
+      ],
+      // decode from the first 9 words
+      eventData.substring(0, 64 * 9 + 2)
     )
+    const addressFromBigNumber = (bn: BigNumber) =>
+      getAddress(hexZeroPad(bn.toHexString(), 20))
+    const destAddress = addressFromBigNumber(parsed[0])
+    const l2CallValue = parsed[1] as BigNumber
+    const l1Value = parsed[2] as BigNumber
+    const maxSubmissionFee = parsed[3] as BigNumber
+    const excessFeeRefundAddress = addressFromBigNumber(parsed[4])
+    const callValueRefundAddress = addressFromBigNumber(parsed[5])
+    const gasLimit = parsed[6] as BigNumber
+    const maxFeePerGas = parsed[7] as BigNumber
+    const data = '0x' + eventData.substring(64 * 9 + 2)
```

Comment:
don't we want to use the parsed data length here?
something like
```ts
const dataLength = parsed[8] as BigNumber
const data = '0x' + eventData.substring(
    eventData.length - dataLength.toNumber()
)
```
I think we'd also need to adjust `eventData.length - dataLength.toNumber()` to take into account padding